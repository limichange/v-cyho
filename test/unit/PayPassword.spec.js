import { shallow } from '@vue/test-utils'
import PayPassword from '../../src/PayPassword.vue'
import { createRenderer } from 'vue-server-renderer'

describe('PayPassword.vue', () => {
  it('default render', () => {
    const wrapper = shallow(PayPassword)
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('set length as 10', () => {
    const wrapper = shallow(PayPassword, {
      propsData: {
        length: 10
      }
    })

    expect(wrapper.findAll('i')).toHaveLength(10)
  })

  it('set value as "123456"', () => {
    const value = '123456'
    const wrapper = shallow(PayPassword, {
      propsData: {
        value
      }
    })

    expect(wrapper.vm.val).toBe(value)
  })

  it('clear value', () => {
    const value = '123456'
    const wrapper = shallow(PayPassword, {
      propsData: {
        value
      }
    })

    wrapper.vm.clear()
    expect(wrapper.vm.val).toBe('')
  })
})
