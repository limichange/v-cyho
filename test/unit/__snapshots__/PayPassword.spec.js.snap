// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayPassword.vue default render 1`] = `
<span class="payPassword"><input type="password" name="payPassword" autocomplete="off" maxlength="6" minlength="6" value="" class="sixDigitPassword"> <div tabindex="0" class="sixDigitPassword"><i><b style="display:none;"></b></i><i><b style="display:none;"></b></i><i><b style="display:none;"></b></i><i><b style="display:none;"></b></i><i><b style="display:none;"></b></i><i><b style="display:none;"></b></i> <span style="width:29px;left:0px;display:none;"></span></div>
</span>
`;

exports[`PayPassword.vue matches snapshot 1`] = `
<div tabindex="0" class="sixDigitPassword">
  <i><b style="display: none;"></b></i>
  <i><b style="display: none;"></b></i>
  <i><b style="display: none;"></b></i>
  <i><b style="display: none;"></b></i>
  <i><b style="display: none;"></b></i>
  <i><b style="display: none;"></b></i>
  <span style="width: 29px; left: 0px; display: none;"></span>
</div>
`;
